pkg/connections/type.go

```diff
@@ -0,0 +1,72 @@
+package connections
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/absmach/magistrala/pkg/errors"
+)
+
+var errInvalidConnType = errors.New("invalid connection type")
+
+type Type uint8
+
+const (
+	Invalid Type = iota
+	Publish
+	Subscribe
+)
+
+func (c *Type) UnmarshalJSON(bytes []byte) error {
+	var cstr string
+	if err := json.Unmarshal(bytes, &cstr); err != nil {
+		return err
+	}
+
+	nc, err := StringToType(cstr)
+	if err != nil {
+		return err
+	}
+	*c = nc
+	return nil
+}
+
+func (c Type) CheckType() error {
+	switch c {
+	case Publish:
+		return nil
+	case Subscribe:
+		return nil
+	default:
+		return fmt.Errorf("Unknown connection type %d", c)
+	}
+}
```

Comment:
let's not attach this to `Type`, but make this an exported function `CheckType(t Type) error`.